FirebaseSessions/Sources/SessionGenerator.swift

```diff
@@ -43,11 +43,23 @@ class SessionGenerator {
 
   private var firstSessionId = ""
   private var sessionIndex: Int32
+  private var collectEvents: Bool
 
-  init(settings: SettingsProtocol) {
+  init(settings: SettingsProtocol,
```

Comment:
Should this change be here or should this move to FirebaseSessions initializer? I'm thinking this logic of checking if sessions can be generated can be moved to [here](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseSessions/Sources/FirebaseSessions.swift#L145). This is before listening the lifecycle events.

The only catch is that we might have to update settings before the listening begins. If that is done, then I think that would be a much more ideal route. What do you think?

Reply:
I like that idea from a simplicity perspective. My only problem with doing that is it's going to involve a pretty big refactor to the tests

Reply:
Hmm actually there might be a simple way to do it